src/plugins/mer/merhardwaredevicewizardpages.cpp

```diff
@@ -135,20 +139,49 @@ void MerHardwareDeviceWizardSelectionPage::handleTestConnectionClicked()
     sshParams.timeout = timeout();
     sshParams.authenticationType = SshConnectionParameters::AuthenticationTypeAll;
 
+    QString errorMessage;
     m_ui->connectionTestLabel->setText(tr("Connecting to machine %1 ...").arg(hostName()));
     m_ui->connectionTestLabel->setText(MerConnectionManager::testConnection(sshParams,
                 &m_connectionTestOk));
     if (!m_connectionTestOk)
         goto end;
 
     m_ui->connectionTestLabel->setText(tr("Detecting device properties..."));
-    m_architecture = detectArchitecture(sshParams, &m_connectionTestOk);
-    m_deviceName = detectDeviceName(sshParams, &m_connectionTestOk);
+    m_architecture = detectArchitecture(sshParams, &m_connectionTestOk, &errorMessage);
+    m_deviceName = detectDeviceName(sshParams, &m_connectionTestOk, &errorMessage);
     if (!m_connectionTestOk) {
         m_ui->connectionTestLabel->setText(tr("Could not autodetect device properties"));
+        showErrorMessageDialog(tr("Could not autodetect device properties."), errorMessage);
         goto end;
     }
 
+    m_ui->connectionTestLabel->setText(tr(
+        "Detecting software packages required for proper SDK operation..."));
+    m_sdkClientToolsInstalled = detectSdkClientTools(sshParams, &m_connectionTestOk, &errorMessage);
+    if (!m_connectionTestOk) {
+        m_ui->connectionTestLabel->setText(tr("Could not autodetect installed software packages"));
+        showErrorMessageDialog(tr(
+            "Could not autodetect installed software packages for proper SDK operation."), errorMessage);
+        goto end;
+    }
+    if (!m_sdkClientToolsInstalled) {
+        m_ui->connectionTestLabel->setText(tr(
+            "Software packages required for proper SDK operation missing"));
```

Comment:
I would write "*are* missing" in this case, what do you think?

Reply:
Yeah, that's better and fits still. (At least on my computer!)